File: Baekjoon/문자열/10610.py
# https://www.acmicpc.net/problem/10610

n = list(input())
answer = -1

max_num = sorted(n, reverse = True)
max_num = int(''.join(max_num))

if max_num % 30 == 0:
    answer = max_num

print(answer)

File: Baekjoon/문자열/10610.cpp
// https://www.acmicpc.net/problem/10610

#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

int main() {
    string N;
    cin >> N;

    int sum = 0;
    bool hasZero = false;
    for (char c : N) {
        sum += c - '0';
        if (c == '0') {
            hasZero = true;
        }
    }

    if (sum % 3 == 0 && hasZero) {
        sort(N.begin(), N.end(), greater<char>());
        cout << N << '\n';
    } else {
        cout << -1 << '\n';
    }

    return 0;
}